src/pyk/proof/proof.py

```diff
@@ -176,9 +176,21 @@ def subproofs_status(self) -> ProofStatus:
 
     @property
     @abstractmethod
-    def status(self) -> ProofStatus:
+    def own_status(self) -> ProofStatus:
         ...
 
+    @property
+    def status(self) -> ProofStatus:
+        if self.admitted:
+            return ProofStatus.PASSED
+
+        if self.own_status == ProofStatus.FAILED or self.subproofs_status == ProofStatus.FAILED:
+            return ProofStatus.FAILED
+        elif self.own_status == ProofStatus.PENDING or self.subproofs_status == ProofStatus.PENDING:
+            return ProofStatus.PENDING
+        else:
+            return ProofStatus.PASSED
+
```

Comment:
```suggestion
    @property
    def status(self) -> ProofStatus:
        if self.admitted:
            return ProofStatus.PASSED
        if self.own_status == ProofStatus.FAILED or self.subproofs_status == ProofStatus.FAILED:
            return ProofStatus.FAILED
        elif self.own_status == ProofStatus.PENDING or self.subproofs_status == ProofStatus.PENDING:
            return ProofStatus.PENDING
        else:
            return ProofStatus.PASSED

```

Also probably don't need `elif` logic here, since they're all just returning. But otherwise looks good.

Reply:
I'm not sure why the diff is all crazy, I just deleted a blank line.